lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/spatial/Edge.java

```diff
@@ -146,4 +146,24 @@ public Vector3d getNearestPointOnEdge(Edge<?> otherEdge, Vector3d result) {
     public double getLength() {
         return a.distanceTo(b);
     }
+
+    public boolean isIntersectingEdge(Edge<Vector3d> otherEdge) {
+        Vector3d r = new Vector3d(a.x - b.x, a.y - b.y, 0);
```

Comment:
does this even work? in Vector3d the y-axis is representing the altitude, this is NOT equal to the y-axis of an CartesianPoint. When you call `new CartesianPoint(3,4,0).toVector3d()`, then the resulting Vector3d would contain : `x=3, y=0, z=-4` (see `MutableCartesianPoint.toVector3d()`, This is because the axes have different meanings in phabmacs.

Reply:
I would just move this code to CartesianPolygon as proposed in my previous comment and work with CartesianPoints only.

Reply:
Weirdly it always worked, but it also seems to work now with the coordinates used correctly. I will check that more thoroughly.